### PR TITLE
[+] Added implementation of UserJoinUseCaseTest.

### PR DESCRIPTION
**Pull request by xgao28**
In this pull request, I initially implemented the UserJoinUseCaseTest based on three different scenarios that the UserJoinUseCase may encounter depending on the user input and data in the database.

### CODE REVIEW

**Review by ScottCTD — DISMISSED** [review]
The tests passed! COOL!

File: src/test/java/billgates/use_cases/UserJoinUseCaseTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import billgates.interface_adapters.DatabaseGateway;
+import billgates.use_cases.user_join.*;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * A test for <code>UserJoinUseCase</code>.
+ * This test utilizes
```
Comment (Lei-Tin) [inline src/test/java/billgates/use_cases/UserJoinUseCaseTest.java:14]: Can we write something here, like

This test utilizes the User with the user ID "243545" and a bill ID of "756432"

File: src/test/java/billgates/use_cases/UserJoinUseCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+public class UserJoinUseCaseTest {
+
+    private DatabaseGateway gateway = new MySQLDatabaseGateway();
+    private UserJoinController controller;
+    private UserJoinPresenter presenter;
+    private UserJoinUseCase useCase;
+    private UserJoinViewModel viewModel;
+    private final User user = User.createInstance(243545, "userjoin", "usecasetest", 756432);
```
Comment (Lei-Tin) [inline src/test/java/billgates/use_cases/UserJoinUseCaseTest.java:24]: Maybe setting the 243545 and 756432 as final constant variables may be better, magical numbers in a constructor is a type of code smell.

Also, maybe adding cmoments will help

**Review by Lei-Tin — APPROVED** [review]
Tests seems like passing, good!

**Review by Lei-Tin — APPROVED** [review]
GOOD, APPROVED!

**Review by ScottCTD — APPROVED** [review]

**Pull request merged**